### PR TITLE
Entry list

### PR DESCRIPTION
**Pull request by donacrisante**

### CODE REVIEW

**Comment by donacrisante** [comment]
Quick info: I also added a Favourite button/icon, a Favourite Tab and a favourite toggle function in this US. My goal was also to get all entries displayed I you click on the "Recent journeys" tab and only the favourites when I click on the "Favourites" tab. The amount of entries and of the favourites should also be displayed.

File: components/Divider.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    return <hr name="divider" role="none" />;
+  }
+
+  const Hr = styled.hr`
+  border: none;
+  width: 100%;
+  margin: 0;
+  height: 1px;
+  background: color-water-10;
+`;
```
Comment (ninaassmann) [inline components/Divider.js:13]: you make a styled hr here but you are using a normal one
i guess you need to change the hr in the return to Hr

File: components/EntryList.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+              id={entry.id}
+              isFavourite={entry.isFavourite}
+              onToggleFavourite={onToggleFavourite}
+            />
+            <p>{entry.date}, {entry.start} - {entry.destination}, {entry.km}km, {entry.transport} {entry.fuel}, {result} kg CO<sub>2</sub></p>
+          </Fragment>
+        ))}
+      </div>
+      </section>
+      <button type="submit"> + </button>
```
Comment (ninaassmann) [inline components/EntryList.js:52]: this button does nothing, I guess the form you open
so maybe use a Link component with href="/"
when it should look like a button yu can use a styled component for that

**Review by ninaassmann — CHANGES_REQUESTED** [review]

File: components/Divider.js (hunk 1 above)
Comment (donacrisante) [reply components/Divider.js:13]: Hi Nina, thanks for the feedback! Good point! I had totally overlooked it. Thanks

**Comment by donacrisante** [comment]
I had already sent my PR to the code review on Friday, but since I still discovered an error and have now fixed it, I am sending my PR to the code review again. The error was that my CO2 calculation worked, but the displayed result in the entry list was the same for all modes of transport. It was not updated depending on which means of transport I had specified. Now it should work.

File: components/EntryList.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import Divider from "./Divider";
+import { Fragment } from "react";
+import Tab from "./Tab";
+import Tabs from "./Tabs";
+/* import Badge from "./Badge"; */
```
Comment (Kheirah) [inline components/EntryList.js:5]: Did you accidentally comment this out?
You are using an element called `badge` (in lowercase) in line 26 and 33. You probably want to use the actual `Badge` component for which you would need the import.

File: components/EntryForm.js [hunk 4]
```diff
@@ -50,23 +22,59 @@ export default function EntryForm() {
...
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const newEntry = Object.fromEntries(formData);
+    const transport = newEntry.transport;
+    const km = newEntry.km;
+    const fuel = newEntry.fuel;
+    const result = handleCalculateCo2(transport, km, fuel);
+    newEntry.result = result;
+    console.log(newEntry);
+    console.log(result);
```
Comment (Kheirah) [inline components/EntryForm.js:55]: Remove `console.logs` when you're done inspecting the behavior of your code.

File: pages/api/hello.js
Comment (Kheirah) [inline pages/api/hello.js:1]: You can delete this file. It's included in any new project, but you don't need it.

File: pages/_app.js [hunk 5]
```diff
@@ -1,14 +1,63 @@
...
+    defaultValue: [],
+  });
+
+  const [filter, setFilter] = useLocalStorageState("filter", {
+    defaultValue: "all",
+    });
+
+  const [isFavourite, setIsFavourite] = useLocalStorageState("favourite", {
+    defaultValue: "false",
+  });
```
Comment (Kheirah) [inline pages/_app.js:19]: Mhh, are you sure you need this state? I can't see where you are using it. And you don't need it to be honest. I think you can remove it and remove `isFavorite` from the being passed down via props as well.

**Review by Kheirah — APPROVED** [review]
Very good!

Have a look at the state you're not really using and see if you can remove it since it's probably not necessary to keep around.

File: components/EntryList.js (hunk 3 above)
Comment (donacrisante) [reply components/EntryList.js:5]: Yes, I wanted to test something with this Badge and forgot to uncomment it. I did it now.

File: components/EntryForm.js (hunk 4 above)
Comment (donacrisante) [reply components/EntryForm.js:55]: I have removed them.

File: pages/api/hello.js
Comment (donacrisante) [reply pages/api/hello.js:1]: Done.

Thanks for your feedback.

**Comment by donacrisante** [comment]
I had sent my code to QA this morning, but the reviewer noticed that when I enter a number of kilometers with a comma, the result of the CO2 calculation shows NaN. So, to fix this error I added a regular expression to the Km input field (in EntryForm component) that accepts single numbers and decimal numbers with comma or point. I have also updated the handleCalculateCo2 function accordingly. Since I was already at validation attributes, I also added the attribute "required" and a regular expression for the two other input fields (start, destination).

**Review by Kheirah — APPROVED** [review]
Nice!

Patterns are a good way to check the entered value in inputs, **but only when the submit event is fired.**

Currently, you have a "calculate" button which _does not check_ the entered value. It's okay for now because I know you're planning to remove that button anyway. And then you're left with the submit button only which will check the values when clicked.

**Pull request merged**